src/app/modules/training-event/components/table-list/table-list.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { MatPaginator, PageEvent } from '@angular/material';
+import { of as observableOf, Subject, merge } from 'rxjs';
+import { switchMap, map, catchError } from 'rxjs/operators';
+import { CampusEventResponse } from 'src/app/interfaces/event';
+import { EventService  } from '../../services/event.service';
+
+@Component({
+  selector: 'app-table-list',
+  templateUrl: './table-list.component.html',
+  styleUrls: ['./table-list.component.css']
+})
+export class TableListComponent implements OnInit {
```

Comment:
This is a bad name considering that the name is general and not related to specific model. We can discuss whether it's appropriate to use a general component to display list of instances for all models, but here let's name it as `CampusEventListComponent`.